llvm/lib/Target/NVPTX/NVPTXSubtarget.h

```diff
@@ -118,6 +119,8 @@ class NVPTXSubtarget : public NVPTXGenSubtargetInfo {
 
   NVPTXSubtarget &initializeSubtargetDependencies(StringRef CPU, StringRef FS);
   void ParseSubtargetFeatures(StringRef CPU, StringRef TuneCPU, StringRef FS);
+
+  void requireClusters(std::string const &FailureMessage) const;
```

Comment:
We need to make clear that this produces a fatal error if conditions are not met.

`require` alone does not carry "I am a fatal error" vibe.  
I think we need something painfully obvious, along the lines of `failIfClustersNotAvailable()`.

Reply:
Done